diff --git a/bylaws.current b/bylaws.current
index dd8d801..9f77eb0 100644
--- a/bylaws.current
+++ b/bylaws.current
@@ -437,45 +437,230 @@ membership of all ordinary committees and of the ownership of all
 extraordinary committees.
 
                               ARTICLE IX
-                  REPRIMAND, SUSPENSION, AND EXPULSION
-
-Section 1. Any member of the organization shall be held accountable for
-any conduct considered injurious to the organization or its purposes
-and objectives. The gravity of such conduct shall first be considered
-at a hearing before the Executive Board.  Considering said charges of
-conduct, the Board may, at its discretion, enforce any of the following
-punitive actions:
-
-    a. Reprimand. The Board may reprimand the member for minor offenses,
-    which can include, but shall not be limited to, written reprimand,
-    verbal warning, restitution, or organization service.
-
-    b. Suspension. For offenses of a more serious nature, the Board may
-    suspend membership privileges of the member for any length of time,
-    up to the next Annual Meeting of the organization.
-
-    c. Expulsion. Considering the most heinous of offenses, the Board may
-    propose to the organization membership expulsion of the member. The
-    organization shall then vote at the next Regular Meeting of the
-    organization, and the proposal for expulsion shall require a
-    two-thirds (2/3) vote for approval.
-
-Section 2. Any member so charged shall be given written notice of said
-hearing at least ten (10) days prior to that hearing, and such member
-shall have the right to be present at said hearing for the purposes
-of defense.
-
-Section 3. Any member affected by Section 1, Parts (a) and (b), of this
-Article, shall have thirty (30) days to appeal said proceedings. Written
-notice of appeal shall be presented to the Recording Secretary. The
-Board shall then, in turn, present the matter to the voting membership
-at the next Regular organization meeting.  An affirmative two-thirds
-(2/3) vote shall be required to sustain the actions of the Board.
-
-Section 4. Any person whose membership is terminated according to Section
-1 of this article must present his or her application for reinstatement
-to the Executive Board for approval prior to being voted upon by the
-general membership.
+           REPRIMAND, PROBATION, SUSPENSION, AND EXPULSION
+
+Section 1. Submitting a Complaint. Any person affiliated with the
+organization can be held accountable for any conduct deemed
+inappropriate or harmful to the organization, its purposes,
+objectives, or affiliated people. Any person affiliated with the
+organization may, at any time, issue a complaint against any other
+person affiliated with the organization by contacting any member of
+the Executive Board in any way, including but not limited to verbal or
+written communication. The Executive Board may also issue a complaint
+about any person affiliated with the organization.
+
+    a. Complaints may be about actions or events that occurred in any
+    context. This includes, but is not limited to, interactions online, in
+    person, and at organization events.
+
+Section 2. Involvement of Officers. If a complaint is issued against
+an officer, that officer must withdraw themself from all Executive
+Board activity related to that complaint. Any officer may also choose
+to withdraw themself from such activity at any moment and for any
+other reason.
+
+    a. If quorum of the Board cannot be obtained with the remaining
+    officers the complaint and the process fall under Section 7.
+
+    b. For the remainder of this article the Executive Board and officers
+    refer to those who have not withdrawn themselves, except in deciding
+    quorum of the Board.
+
+Section 3. Starting the Procedure. Within one (1) week after a
+complaint has been issued, the Executive Board must do the following:
+
+    a. The Board must offer to hold separate discussions with the accused
+    party and the accuser in order to determine the severity of the
+    situation. Either party may refuse this meeting.
+
+        1) All meetings with any involved parties must take place with at
+        least two (2) officers, one of whom must be the President or most
+        senior officer not withdrawn as defined in Section 2.b. Each party
+        may request to include a specific member of the Board at the
+        meeting. Each person may also have a third party present at the
+        meeting, subject to advance approval of the Board.
+
+        2) Any meeting held with an accused party may be held in the
+        presence of a representative of the Student Activities team within
+        the Office of Student Leadership, Involvement and Civic Engagement
+        (hereafter, “Student Activities”), at the discretion of the
+        Executive Board.
+
+    b. If the Executive Board believes that the complaint may require
+    disciplinary action beyond a Reprimand (i.e., Probation, Suspension,
+    or Expulsion), they must inform a representative from Student
+    Activities of the situation. The Executive Board does not need to
+    receive a response before issuing the disciplinary action.
+
+The Executive Board may then choose to take one of the disciplinary
+actions defined in Section 4.
+
+Section 4. Disciplinary Actions. In response to a complaint, the
+Executive Board may issue any of the following disciplinary actions
+against the accused party. The Executive Board must approve such an
+action by majority vote, unless specified otherwise. If disciplinary
+action is approved, the Board must send written notice of such action
+to the accused party within 24 hours of the vote.
+
+    a. Reprimand. The Board may issue a reprimand to the accused party for
+    minor offenses. The reprimand will take the form of a written notice
+    and a verbal warning including, but not limited to, a warning of
+    future consequences should the accused party continue their behavior.
+    The verbal warning will be given during a meeting between the accused
+    party and at least one member of the Executive Board.
+
+    b. Probation. For moderate offenses that have the potential to
+    escalate, the Board may place the accused party on probation. The
+    accused party may still participate in the activities of the
+    organization, but must adhere to the conditions of a written
+    agreement. The agreement:
+
+        1) Will be written by the Executive Board in discussion with a
+        representative from Student Activities, with input from the accused
+        party.
+
+        2) Will include a set of conditions (e.g., restricted actions and
+        corresponding consequences) that the accused must adhere to.
+
+        3) May restrict the accused party from attending some, but not all,
+        organization activities.
+
+        4) Will include the following information:
+
+            A) If the Board determines that the conditions of the agreement
+            are broken in any way, the Executive Board may immediately
+            escalate the disciplinary action to Suspension or Expulsion.
+
+            B) The Executive Board may end a Probation at any time at their
+            discretion.
+
+        5) Will be signed by the President and the accused party.
+
+    c. Suspension. For offenses of a more serious nature, the Board may
+    forbid the accused party from attending all organization activities,
+    and, if applicable, suspend the accused party’s membership privileges.
+
+        1) Upon receiving a complaint, the Executive Board may issue a
+        Temporary Suspension, which is a full ban from all organization
+        activities, for up to one (1) week. The Executive Board does not have
+        to meet the requirements in Section 3 to enact a Temporary Suspension.
+        Temporary Suspensions may be enacted once per complaint and cannot be
+        extended.
+
+        2) Upon issuing a Suspension, the Board:
+
+            A) Must specify its duration. If the Suspension extends beyond the
+            next Annual Meeting, each new Executive Board must, during its
+            first meeting, decide by majority vote to uphold or end the
+            Suspension.
+
+            B) May set objectives for the suspended party during the term of
+            their Suspension. The Executive Board will define these objectives
+            in discussion with the accused party and a representative from
+            Student Activities. Completion of these objectives will end the
+            suspension as specified in Section 4.c.3.
+
+            C) Must provide written notice to the suspended party of the
+            duration of the suspension, any objectives set for the suspension,
+            and the procedures given in Sections 4.c.3, 4.c.4, and 4.c.5.
+
+        3) The Executive Board may end a Suspension at any time by majority
+        vote. If the Board determines that the suspended party has
+        accomplished the objectives set forward in Section 4.c.2.B, the Board
+        must immediately end the Suspension.
+
+        4) A party whose Suspension has ended will be placed on probation. The
+        Executive Board must meet to define the terms of the probation, as
+        defined in Section 4.b.
+
+        5) If a Suspension is violated, the Executive Board may modify the
+        Suspension or immediately move to expel the suspended party.
+
+    d. Expulsion. For heinous offenses, the Board may move to
+    expel the accused party from the organization. To issue an expulsion:
+
+        1) The Board must hold a meeting with a representative of Student
+        Activities to discuss the motion to expel. All non-withdrawn officers
+        must attend, and a written statement that the meeting was held must be
+        signed by all present, including the representative of Student
+        Activities.
+
+        2) After this meeting is held, any non-withdrawn officers may recuse
+        themselves from voting on the motion. If quorum of the Board cannot be
+        obtained with the remaining officers, the motion cannot proceed.
+
+        3) The Board will then vote on the motion to expel. No officer may
+        abstain. If there are two (2) or more “no” votes, the motion fails.
+        Otherwise, it passes and the expulsion goes into effect.
+
+    The Executive Board must immediately provide written notice of the
+    expulsion to the expelled party. Once the Board has informed the
+    expelled party, it must not discuss the expulsion with the general
+    body until twenty-four (24) hours have passed. Within one (1) week of
+    the expulsion, the Board must send a mailing to the body stating that
+    an expulsion has occurred. The Board must disclose the identity of the
+    expelled party to any member who asks, but may choose not to announce
+    it to the general body. The Board must present the written
+    confirmation from Student Activities to any member, or to the expelled
+    party, upon request.
+
+Section 5. Appeals of Probation or Suspension. Any party against whom
+the Executive Board has issued a Probation or Suspension may appeal
+the decision of the Board within thirty (30) days of the Board issuing
+the disciplinary action. The affected party must present their appeal
+to the Executive Board in writing. Between 7 and 10 days after
+receiving the appeal, the Board must begin the appeals process, as
+defined in Section 5.b or Section 5.c.
+
+    a. With their appeal, the affected party must request either for the
+    appeal to be mediated by Student Activities or voted on by the general
+    body. The accuser, or any person who has been adversely affected by
+    actions related to the complaint, may also make such a request. If not
+    all such requests are in agreement, the appeal will be mediated by
+    Student Activities.
+
+    b. If the appeal is to be mediated by Student Activities, the affected
+    party will meet, as defined in Section 3.a, with a representative from
+    Student Activities and two (2) officers, to resolve the appeal.
+
+    c. If the appeal is to be voted on by the general body, the Recording
+    Secretary will mail notice of the appeal to the membership. The Board
+    will then present the matter to the voting membership at the first
+    Regular Meeting at least six (6) days after the mailing. An
+    affirmative two-thirds (2/3) vote will be required to sustain the
+    actions of the Board.
+
+    d. Any given decision of the Board can only be appealed once.
+
+Section 6. Reinstatement after Expulsion. Any person who has been
+expelled may apply to have their expulsion be repealed.
+
+    a. This application must be sent in writing to the Board. Within one
+    week, the Board. Within one week, the Board will vote to approve the
+    application. The application can only be approved by a unanimous vote.
+    If the Board approves the application, the Recording Secretary will
+    mail notice of the application to the membership within twenty-four
+    (24) hours of the Board’s decision. If the Board rejects the
+    application, they will send written notice of this decision to the
+    applicant within twenty-four (24) hours.
+
+    b. The Board will then present the matter to the voting membership at
+    the first Regular Meeting at least six (6) days after the mailing.
+
+    c. If the expelled party so desires, they may submit a written defense
+    which a representative of the Executive Board will read to the general
+    body during the discussion prior to the vote. This defense is not
+    required for the vote to commence.
+
+    d. An affirmative two-thirds (2/3) vote will be required to repeal the
+    expulsion.
+
+Section 7. Extenuating Circumstances. If the procedures in this
+Article are not appropriate or have ceased to be sufficient for a
+given complaint, the Executive Board will meet with a representative
+from Student Activities to develop a process which is in line with the
+values and intentions of the organization and these bylaws.
+
 
                                ARTICLE X
                          PARLIAMENTARY AUTHORITY
